packages/amplify-provider-awscloudformation/src/iterative-deployment/helpers.ts

```diff
@@ -0,0 +1,50 @@
+import { DeployMachineContext, DeploymentMachineOp } from './state-machine';
+import * as path from 'path';
+
+export const hasMoreRollback = (context: DeployMachineContext) => {
+  return context.currentIndex >= 0;
+};
+
+export const hasMoreDeployment = (context: DeployMachineContext) => {
```

Comment:
I would reverse the logic here and rename to `isDeploymentComplete`. Or if you don't want to reverse the logic, rename to `isDeploymentIncomplete`. Same with `hasMoreRollback` -> `isRollbackComplete`